torch/_functorch/aot_autograd.py

```diff
@@ -776,225 +777,336 @@ def unpack_synthetic_bases(
             f_args_inner.append(view_arg)
     return f_args_inner
 
-# This creates the final function that we want to trace using make_fx(),
-# in both aot_dispatch_autograd and aot_dispatch_base.
-# The function returns has signature "traced_fn(primals: List[Any], maybe_tangents: Optional[List[Any]])",
-# where:
-# (1) if trace_joint is True, it will trace the joint fwd/bwd (by calling autograd.grad on the result of the forward)
-# (2) if trace_joint is False, it will only trace the forward.
-# It uses the metadata that was created earlier to figure out what all of the outputs to the autograd.Function.forward are:
-# (1) Which inputs received data mutations (and need to be passed as outputs into autograd.grad())
-# (2) Which outputs are aliases of inputs (and should *not* be passed as outputs into autograd.grad())
-def create_forward_or_joint_functionalized(
-    fn,
-    *,
-    meta: ViewAndMutationMeta,
-    synthetic_base_info: Optional[List[Union[int, Tuple[int, torch.Tensor]]]],
-    trace_joint: bool,
-    keep_input_mutations: bool
-):
-    # What's happening here? For any inputs in the graph that are mutated, we need to clone them first
-    # (and similarly for metadata-only mutations, we need to view them first).
-    # The idea is that when we trace the backward, we need to pass in the *original* primals
-    # to autograd.grad(), before they were mutated.
-    #
-    # NOTE: when we have synthetic base inputs, we need to clone them *before* creating views off of them.
-    # This means that "idx" here represents the index of the (potentially) synthetic base.
-    # What we need to do is:
-    # (1) map the current (post-synthetic-base calling convention) input argument index
-    #     to int index pre-synthetic-base-calling-convention.
-    # (2) There could be multiple, if this index corresponds to a synthetic base
-    #     that has multiple input aliases.
-    # (3) If any of those corresponding inputs get metadata mutations, then we clone the base.
-    def maybe_to_fresh_input(idx, t):
-        if not isinstance(t, Tensor):
-            return t
+# This class contains all the metadata we care about for the current function we're compiling.
+# This data is needed both at trace time and at runtime.
+@dataclass
+class CompiledRuntimeMetadata:
+    # This type / object should be cleaned up
+    synthetic_base_info: Optional[List[Union[int, Tuple[int, torch.Tensor]]]]
```

Comment:
xref this with where the info is documented

Reply:
done